packages/babel-preset-gatsby/src/index.js

```diff
@@ -70,6 +70,7 @@ module.exports = function preset(_, options = {}) {
       ],
     ],
     plugins: [
+      resolve(`babel-plugin-optimize-hook-destructuring`),
```

Comment:
Next.js is using `lib: true` option (which limit this optimization to hooks imported from `react` or `preact/hooks`) - https://github.com/zeit/next.js/blob/ce3f7d05b30520dfd37879b0b09f9f6109fa2367/packages/next/build/babel/preset.ts#L134-L140

I think that's safe thing to do here as well?